Onboarding: Beginner lecture (+ runnable Mesh examples) (#1916)
cardano-foundation/developer-portal: docs/developers/onboarding/lectures/introduction.md

Reviewer: The Next button at the end links to this same page

---
title: "Lectures"
sidebar_label: "Introduction"
description: "A guided, hands-on path to learning Cardano development, from the basics to high-quality smart contracts."
---

# Lectures

> The Lectures are a guided, hands-on path to learning Cardano development. Each lecture try to explain one idea in plain language with a real-world comparison, and ends with something you can **run and see** for yourself. They come in three modules, take them in order.

Reviewer: ```suggestion
The Lectures are a guided, hands-on path to learning Cardano development. Each lecture tries to explain one idea in plain language with a real-world comparison, and ends with something you can **run and see** for yourself. They come in three modules; take them in order.
```

## The three modules

> ### Beginner: the base of Cardano

Reviewer: ```suggestion
### Beginner: Cardano's Fundamentals
```
wdyt? "the base" doesn't sound native english

The core ideas you need before building anything: wallets, UTxOs, transactions, tokens, and the tools you use to read the chain (providers and explorers). After this you'll understand how Cardano works and be ready to build.

**[Start the Beginner module](/docs/developers/onboarding/lectures/beginner/introduction)**

> ### Intermediate: smart contracts

Reviewer: ```suggestion
### Intermediate: Smart contracts
```

> Smart contracts from scratch: on-chain vs off-chain, what a validator is, datum and redeemer, the languages you write them, and the off-chain SDKs that drive them. After this you'll be able to **write and understand** a smart contract. 

Reviewer: ```suggestion
Smart contracts from scratch: Their structure, their role, on-chain vs off-chain, what a validator is, datum and redeemer, the languages you write them in, and the whole development cycle. After this, you'll be able to **write and understand** smart contracts in Cardano. 
```

> ### Advanced: high-quality smart contracts

Reviewer: ```suggestion
### Advanced: Production-ready smart contracts
```

> Going from "it works" to "it's safe": common vulnerabilities, design patterns, testing, optimization, and getting to production. After this you'll be able to write **secure, high-quality** contracts.

Reviewer: ```suggestion
Going from "it works" to "it's safe and scalable": common vulnerabilities, design patterns, optimization, and getting to production. After this, you'll be able to write **secure, scalable, high-quality** contracts.
```

## What you need

Most lectures have two kinds of hands-on.

**To read and explore (no install).** A modern browser and a wallet, we use **[Lace](https://www.lace.io/)** switched to the **Preview** test network, plus free test ADA from the **[faucet](https://docs.cardano.org/cardano-testnets/tools/faucet)**. Preview coins have no real value, so you can experiment safely.

> **To run the code snippets.** Some lectures show real, tested code you can run:

Reviewer: These bullet points mix required tools with steps. Let's do only the required tools and put the steps when they're needed

> - **[Node.js](https://nodejs.org/) latest**, it runs the TypeScript examples directly.

Reviewer: ```suggestion
- Latest **[Node.js](https://nodejs.org/):** It runs the TypeScript examples directly.
```

- **[git](https://git-scm.com/)** to get the example projects, they live under `examples/onboarding/` in the [developer-portal repo](https://github.com/cardano-foundation/developer-portal).
- Then, in an example folder, run `npm install` and `npm test`.

> **Sometimes optional.**

Reviewer: This is very tool-specific. We should either move this to the lectures or add tabs with the required tools, depending on the chosen tools
